Dockerfile

```diff
@@ -13,6 +13,7 @@ RUN apt-get update && apt-get -y install git \
            unzip
 
 RUN mkdir /JavaPrograms
+RUN mkdir /JavaPrograms/scripts
 RUN mkdir /code
```

Comment:
try typing:

```
docker images
```
and you'll see all the (top level) images.. then try

```
docker images --digests
```
and you'll see each image id is actually a sha256 sum! But it gets even better... do

```
docker images --all
```
and you'll see that each "image" is actually a crapton of layers, and they are only assembled when the image is actually running. Each layer is actually a diff off of the previous ones, and then at runtime the top layer is the only one that is actually writable that maintains your changes. This is a fundamentally different model than singularity, which is an actual, one file deal, image.